connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java

```diff
@@ -89,8 +91,22 @@ private ResponseEntity<?> processWebhook(
       ActiveInboundConnector connector, WebhookProcessingPayload payload) {
     ResponseEntity<?> connectorResponse;
     try {
-      var webhookResult =
-          ((WebhookConnectorExecutable) connector.executable()).triggerWebhook(payload);
+      WebhookConnectorExecutable connectorHook =
+          (WebhookConnectorExecutable) connector.executable();
+
+      if (connectorHook.verificationExpression() != null) {
+        var verificationResult = connectorHook.verify(payload);
+        if (verificationResult != null) {
+          HttpHeaders headers = new HttpHeaders();
+          Optional.of(verificationResult)
+              .map(VerifiableWebhook.WebhookHttpVerificationResult::headers)
+              .ifPresent(map -> map.forEach(headers::add));
+          return new ResponseEntity<>(
+              verificationResult.body(), headers, verificationResult.statusCode());
```

Comment:
Lets use single return statements and I would recommend to introduce a dedicated method here.

Reply:
This is fail-fast exit, in case validation failed. In that case, there is no sense to follow down the method. If we want single return statement, it adds up branching in code, which is already too much in this class.

Reply:
> it adds up branching in code, which is already too much in this class.

A good indicator to split up the method 😄

Reply:
Splitted it up. Please take a look.